### PR TITLE
Fix aten implementation for low memory max_pool2d

### PR DESCRIPTION
**Pull request by isuruf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131717

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by peterbell10 — COMMENTED** [review]
Can you add a test?

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -3764,6 +3764,35 @@ def fn2(a):
...
+            indices = prims._low_memory_max_pool2d_offsets_to_indices(
+                offsets,
+                kernel_size[1],
+                x.size(-1),
+                stride,
+                padding,
+            )
+            return vals, indices, offsets
+
+        self.common(fn, (torch.randn(1, 3, 10, 10),))
```
Comment (peterbell10) [inline test/inductor/test_torchinductor.py:3794]: Won't you need to use the aot decomp partition backend to hit the new code?

Comment (isuruf) [reply test/inductor/test_torchinductor.py:3794]: No, this test checks the eager implementation vs the lowering. The new code is used in the eager implementation.

**Review by peterbell10 — APPROVED** [review]

**Pull request closed without merging**